source/usage/other/craftscripts.rst

```diff
@@ -47,54 +45,128 @@ Scripts have the following three variables in their global namespace:
+This returns a new instance of `EditSession <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java>`_. To set blocks, you will need to provide a `BlockVector3 <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector3.java>`_ and then either a `BlockState <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java>`_ or a `BaseBlock <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BaseBlock.java>`_ to the ``EditSession``.
 
-.. topic:: Example: Setting a block to white wool
+
+.. topic:: Example: Setting white wool at (431, 63, -41)
 
     ::
 
-        importPackage(Packages.com.sk89q.worldedit.world.block);
+        importClass(Packages.com.sk89q.worldedit.world.block.BlockTypes);
+        importClass(Packages.com.sk89q.worldedit.math.BlockVector3);
+
+        var editSession = context.remember();
+        editSession.setBlock(BlockVector3.at(431, 63, -41), BlockTypes.WHITE_WOOL.getDefaultState());
+
+
+
+To get blocks, use the function ``EditSession#getBlock``, which accepts a ``BlockVector3`` and returns an instance of ``BaseBlock``.
 
-        var sess = context.remember();
-        sess.setBlock(player.getBlockOn().toVector().toBlockPoint(), BlockTypes.WHITE_WOOL.getDefaultState());
+.. topic:: Example: Replacing white wool at (431, 63, -41) with black wool
 
-Note that because ``BlockTypes`` is in the ``com.sk89q.worldedit.world.block`` namespace, it had to be imported first. The first argument for ``setBlock()`` is a ``BlockVector3`` indicating the position in the world.
+    ::
+
+        importClass(Packages.com.sk89q.worldedit.world.block.BlockTypes);
+        importClass(Packages.com.sk89q.worldedit.math.BlockVector3);
+
+        var editSession = context.remember();
+        var block = BlockVector3.at(431, 63, -41);
+        if (editSession.getBlock(block).getBlockType().equals(BlockTypes.WHITE_WOOL)) {
+          editSession.setBlock(block, BlockTypes.BLACK_WOOL.getDefaultState());
+        }
+
+To get the player's selected region, use:
+
+::
 
-To get blocks, use ``getBlock()`` on ``EditSession``. You'll get back a ``BaseBlock`` too.
+    var region = context.getSession().getRegionSelector(player.getWorld()).getRegion();
 
-Processing arguments
+This returns an instance of `Region <https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java>`_. Although the player cannot run a CraftScript without first selecting a region, the script can read and modify blocks outside the selected region. ``//undo`` will still undo blocks modified outside the selection region.
```

Comment:
>  Although the player cannot run a CraftScript without first selecting a region

This is false. CraftScripts may be run at any time, a selection does not need to be defined.

> ``//undo`` will still undo blocks modified outside the selection region.

I fail to see how this is really relevant here. Just because `//undo` can do something, doesn't really translate into anything related to CraftScripts.

Reply:
2nd point: in MCEdit, a filter that modifies blocks outside the selection region may not save the changes and may not be completely undoable. But this isn't MCEdit so ya. Gonna remove that line.